src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/service/cessations/BeneficialOwnerCessationService.java

```diff
@@ -0,0 +1,101 @@
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+
+    List<Cessation> cessations = new ArrayList<>();
+    cessations.addAll(returnIndividualBeneficialOwners(overseasEntitySubmissionDto, combinedBoData));
+    cessations.addAll(returnCorporateEntityBeneficialOwnerCessations(overseasEntitySubmissionDto, combinedBoData));
+    cessations.addAll(returnLegalPersonBeneficialOwners(overseasEntitySubmissionDto, combinedBoData));
+    return cessations;
+  }
+
+  private List<IndividualBeneficialOwnerCessation> returnIndividualBeneficialOwners(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+    var beneficialOwnersIndividual = overseasEntitySubmissionDto.getBeneficialOwnersIndividual();
+    return beneficialOwnersIndividual.stream()
+            .filter(beneficialOwner -> (beneficialOwner.getCeasedDate() != null))
+            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() != null))
+            .map(beneficialOwner -> returnIndividualBeneficialOwnerCessation(beneficialOwner, combinedBoData))
+            .collect(Collectors.toList());
+  }
+
+  private List<CorporateEntityBeneficialOwnerCessation> returnCorporateEntityBeneficialOwnerCessations(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+    var returned = overseasEntitySubmissionDto.getBeneficialOwnersCorporate();
+    return returned.stream()
+            .filter(beneficialOwner -> (beneficialOwner.getCeasedDate() != null))
+            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() != null))
+            .map(beneficialOwner -> returnCorporateEntityBeneficialOwnerCessation(beneficialOwner, combinedBoData))
+            .collect(Collectors.toList());
+  }
+
+  private List<LegalPersonBeneficialOwnerCessation> returnLegalPersonBeneficialOwners(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+    var returned = overseasEntitySubmissionDto.getBeneficialOwnersGovernmentOrPublicAuthority();
+    return returned.stream()
+            .filter(beneficialOwner -> (beneficialOwner.getCeasedDate() != null))
+            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() != null))
+            .map(beneficialOwner -> returnLegalPersonBeneficialOwnerCessation(beneficialOwner, combinedBoData))
+            .collect(Collectors.toList());
+  }
+
+  private IndividualBeneficialOwnerCessation returnIndividualBeneficialOwnerCessation(
+      BeneficialOwnerIndividualDto bo, Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+
+    var hashedId = bo.getChipsReference();
+    var publicPrivateBoPair = combinedBoData.get(hashedId);
+
+    var appointmentId =  publicPrivateBoPair.getRight().getPscId();
```

Comment:
we have a future story covering "missing" private data. But should probably still filter this case out to avoid NREs.